### DISCUSSION TITLE
Could not find com.android.tools.build:gradle: when executing `./gradlew generateCodegenArtifactsFromSchema`

### DISCUSSION THREAD

**Discussion by silencer07** [reactwg/react-native-new-architecture#200] [Q&A] [answered]
Hi!

I am trying to follow [this](https://dev.to/amazonappdev/a-guide-to-turbo-modules-in-react-native-5aa3?signin=true) tutorial and the https://github.com/reactwg/react-native-new-architecture/blob/main/docs/turbo-modules.md but current I am stuck with this error:

> A problem occurred configuring project ':rtn-splash-screen'.
> Could not determine the dependencies of null.
   > Could not resolve all task dependencies for configuration ':rtn-splash-screen:classpath'.
      > Could not find com.android.tools.build:gradle:.
        > Required by:
            > project :rtn-splash-screen
            
I have tried google and chatgpt and my efforts are in vain. This is the build.gradle of the turbo module

> buildscript {
>   ext.safeExtGet = {prop, fallback ->
>     rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
>   }
>   repositories {
>     mavenCentral()
>     google()
>   }
>   dependencies {
>     classpath("com.android.tools.build:gradle")
>     classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:${safeExtGet('kotlinVersion', '1.9.22')}")
>   }
> }
> 
> apply plugin: 'com.android.library'
> apply plugin: 'com.facebook.react'
> apply plugin: "org.jetbrains.kotlin.android:${safeExtGet('kotlinVersion', '1.9.22')}"
> 
> android {
>   compileSdkVersion safeExtGet('compileSdkVersion', 34)
>   namespace "com.meccabrands.SplashScreen"
> }
> 
> repositories {
>   mavenCentral()
>   google()
> }
> 
> dependencies {
>   classpath("com.android.tools.build:gradle")
>   implementation 'com.facebook.react:react-native'
> }

this is the android folder build.gradle

> buildscript {
>     ext {
>         buildToolsVersion = "34.0.0"
>         minSdkVersion = 30
>         compileSdkVersion = 34
>         targetSdkVersion = 34
>         ndkVersion = "26.1.10909125"
>         kotlinVersion = "1.9.22"
>     }
>     repositories {
>         google()
>         mavenCentral()
>     }
>     dependencies {
>         classpath("com.android.tools.build:gradle")
>         classpath("com.facebook.react:react-native-gradle-plugin")
>         classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlinVersion")
>     }
> }
> 
> subprojects {
>     afterEvaluate { project ->
>         if (project.hasProperty("android")) {
>             project.android.defaultConfig.minSdkVersion = 30
>         }
>     }
> }
> 
> apply plugin: "com.facebook.react.rootproject"

can someone point me out what it is I am doing wrong? Thanks

**Comment by cortinico** [comment]
Please use this template https://github.com/react-native-community/reproducer-react-native to create a reproducer of your failure, and I can then point you in the right direction 👍

**Comment by silencer07** [comment]
Hi @cortinico ,

Thank you very much for extending a helping hand. As requested this is the reproducer repo:

https://github.com/silencer07/codgen-error-reproducer-react-native/tree/main/ReproducerApp

Note: I put the `npx react-native info` output on the said repo as well

**Comment by silencer07** [comment]
Hi @cortinico,

Just wanna bump the discussion. Any insights how to fix this? 

Thanks

**Comment by cortinico** [comment]
Hey @silencer07 my apologies for the late reply but I missed this post.

The problem you have are in the Gradle file of your project:
https://github.com/silencer07/codgen-error-reproducer-react-native/blob/main/ReproducerApp/modules/RTNSplashScreen/android/build.gradle

If you're looking into publishing a library, please update the file with the following content:

```groovy
buildscript {
  ext.safeExtGet = {prop, fallback ->
    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
  }
  repositories {
    google()
    gradlePluginPortal()
  }
  dependencies {
    classpath("com.android.tools.build:gradle:7.3.1")
  }
}

apply plugin: 'com.android.library'
apply plugin: 'com.facebook.react'
apply plugin: 'org.jetbrains.kotlin.android'

android {
  compileSdkVersion safeExtGet('compileSdkVersion', 34)
  namespace "com.meccabrands.SplashScreen"
}

repositories {
  mavenCentral()
  google()
}

dependencies {
  implementation 'com.facebook.react:react-native'
}
```

**Reply by silencer07** [reply]
Hi @cortinico 

I tried copy-pasting and I am still getting error, this time it is more specific.

P.S. I will not publish this. I am just trying to make turbo modules work. I like the idea of it being outside the project, makes migration to expo easier

> * What went wrong:
> A problem occurred configuring project ':rtn-splash-screen'.
> > Could not determine the dependencies of null.
>    > Could not resolve all task dependencies for configuration ':rtn-splash-screen:classpath'.
>       > Could not find com.android.tools.build:gradle:.
>         Required by:
>             project :rtn-splash-screen

I tried changing the jdk version from 17 to 21and vice versa but has no effect

**Reply by cortinico** [reply]
Can you update the reproducer project with your changes? I'll give it another try on Monday

**Reply by silencer07** [reply]
@cortinico I literally copied the code you mentioned here and pushed it to the reproducer app.

Thanks

**Reply by cortinico** [reply]
I've just cloned your repro and the error you mention is gone. The build is failing though as you have a typo in your code (see `ReacTurboModuleoduleInfoProvider`):

```
$ cd ReproducerApp && yarn && yarn android
➤ YN0000: ┌ Resolution step
➤ YN0013: │ rtn-splash-screen@file:./modules/RTNSplashScreen#./modules/RTNSplashScreen::hash=bc3b3d&locator=ReproducerApp%40workspace%3A. can't be found in the cache and will be fetched from
➤ YN0002: │ @react-native/babel-plugin-codegen@npm:0.74.84 doesn't provide @babel/preset-env (p250de), requested by @react-native/codegen
➤ YN0002: │ babel-plugin-transform-flow-enums@npm:0.0.2 doesn't provide @babel/core (p8377f), requested by @babel/plugin-syntax-flow
➤ YN0002: │ react-native@npm:0.74.2 [9aafe] doesn't provide @babel/preset-env (p5949f), requested by @react-native/codegen
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
➤ YN0000: └ Completed
➤ YN0000: ┌ Fetch step
➤ YN0013: │ rtn-splash-screen@file:./modules/RTNSplashScreen#./modules/RTNSplashScreen::hash=bc3b3d&locator=ReproducerApp%40workspace%3A. can't be found in the cache and will be fetched from
➤ YN0000: └ Completed in 0s 265ms
➤ YN0000: ┌ Link step
➤ YN0000: └ Completed
➤ YN0000: Done with warnings in 0s 735ms
info Installing the app...

> Task :rtn-splash-screen:compileDebugKotlin FAILED
49 actionable tasks: 7 executed, 42 up-to-date

info 💡 Tip: Make sure that you have set up your development environment correctly, by running npx react-native doctor. To read more about doctor command visit: https://github.com/react-native-community/cli/blob/main/packages/cli-doctor/README.md#doctor

e: file:///Users/ncor/pg/codgen-error-reproducer-react-native/ReproducerApp/node_modules/rtn-splash-screen/android/src/main/java/com/meccabrands/SplashScreen/DevicePackage.kt:6:40 Unresolved reference: ReacTurboModuleoduleInfoProvider
e: file:///Users/ncor/pg/codgen-error-reproducer-react-native/ReproducerApp/node_modules/rtn-splash-screen/android/src/main/java/com/meccabrands/SplashScreen/DevicePackage.kt:8:1 Class 'DevicePackage' is not abstract and does not implement abstract base class member public abstract fun getModule(p0: String, p1: ReactApplicationContext): NativeModule? defined in com.facebook.react.TurboReactPackage
e: file:///Users/ncor/pg/codgen-error-reproducer-react-native/ReproducerApp/node_modules/rtn-splash-screen/android/src/main/java/com/meccabrands/SplashScreen/DevicePackage.kt:9:3 'getModule' overrides nothing
e: file:///Users/ncor/pg/codgen-error-reproducer-react-native/ReproducerApp/node_modules/rtn-splash-screen/android/src/main/java/com/meccabrands/SplashScreen/DevicePackage.kt:11:46 Unresolved reference: ReactModuleInfoProvider

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':rtn-splash-screen:compileDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction
   > Compilation error. See log for more details

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 10s
error Failed to install the app. Command failed with exit code 1: ./gradlew app:installDebug -PreactNativeDevServerPort=8081
e: file:///Users/ncor/pg/codgen-error-reproducer-react-native/ReproducerApp/node_modules/rtn-splash-screen/android/src/main/java/com/meccabrands/SplashScreen/DevicePackage.kt:6:40 Unresolved reference: ReacTurboModuleoduleInfoProvider
e: file:///Users/ncor/pg/codgen-error-reproducer-react-native/ReproducerApp/node_modules/rtn-splash-screen/android/src/main/java/com/meccabrands/SplashScreen/DevicePackage.kt:8:1 Class 'DevicePackage' is not abstract and does not implement abstract base class member public abstract fun getModule(p0: String, p1: ReactApplicationContext): NativeModule? defined in com.facebook.react.TurboReactPackage
e: file:///Users/ncor/pg/codgen-error-reproducer-react-native/ReproducerApp/node_modules/rtn-splash-screen/android/src/main/java/com/meccabrands/SplashScreen/DevicePackage.kt:9:3 'getModule' overrides nothing
e: file:///Users/ncor/pg/codgen-error-reproducer-react-native/ReproducerApp/node_modules/rtn-splash-screen/android/src/main/java/com/meccabrands/SplashScreen/DevicePackage.kt:11:46 Unresolved reference: ReactModuleInfoProvider FAILURE: Build failed with an exception. * What went wrong:
Execution failed for task ':rtn-splash-screen:compileDebugKotlin'.
> A failure occurred while executing org.jetbrains.kotlin.compilerRunner.GradleCompilerRunnerWithWorkers$GradleKotlinCompilerWorkAction > Compilation error. See log for more details * Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org. BUILD FAILED in 10s.
info Run CLI with --verbose flag for more details.
```

**Reply by silencer07** [reply]
@cortinico 

> I've just cloned your repro and the error you mention is gone.

I fixed the typo, still I am having the same error. I wonder what is your local setup? what JDK version are you using?

P.S. at least on your end it went to the compilation stage, while me I am stuck in cannot find android gradle stuff

**Reply by cortinico** [reply]
I'm on JDK 17:

```
java -version
openjdk version "17.0.9" 2023-10-17 LTS
OpenJDK Runtime Environment Zulu17.46+19-CA (build 17.0.9+8-LTS)
OpenJDK 64-Bit Server VM Zulu17.46+19-CA (build 17.0.9+8-LTS, mixed mode, sharing)
```

Anyway Java version shouldn't make a difference here. There is something going on with your environment setup

**Reply by silencer07** [reply]
that's weird. I mean I am already building react native apps in my current macos laptop.

can you check if there is something amiss in the `npx react-native info` output here

https://github.com/silencer07/codgen-error-reproducer-react-native/tree/main/ReproducerApp#readme

**Reply by cortinico** [reply]
Info looks fine to me.
What I can suggest is to enable GitHub Actions on your reproducer. The CI is already configured and will build the app automatically, so you can tell if there is something going on with your setup

**Reply by silencer07** [reply, marked as answer]
Hi @cortinico,

I enabled github actions in my repo. It complained about lock file that will be updated. I run `yarn install` and for some reason this error went away.

What a relief! I think the error messsage should be improved (i.e. point out the problematic yarn lock file instead of the error in this thread)

Thank you very much. you are such a big help!
